tiledb/api/c_api/vfs/vfs_api_experimental.h

```diff
@@ -93,7 +104,15 @@ TILEDB_EXPORT capi_return_t tiledb_vfs_ls_recursive(
     tiledb_ctx_t* ctx,
     tiledb_vfs_t* vfs,
     const char* path,
-    tiledb_ls_callback_t callback,
+    tiledb_ls_file_callback_t file_callback,
+    void* data) TILEDB_NOEXCEPT;
+
+TILEDB_EXPORT capi_return_t tiledb_vfs_ls_recursive_v2(
+    tiledb_ctx_t* ctx,
+    tiledb_vfs_t* vfs,
+    const char* path,
+    tiledb_ls_file_callback_t file_callback,
+    tiledb_ls_dir_callback_t dir_callback,
     void* data) TILEDB_NOEXCEPT;
```

Comment:
I didn't know that we are going to add a new API for this; my understanding was that we would just pass object storage common prefixes to the file callback, like we already do in the local filesystem.

If we decide to expose the directory filter capability in the C API, we should stop calling the file filter in both the v2 and the existing APIs, but as I said in https://github.com/TileDB-Inc/TileDB/pull/5596#discussion_r2237164369, the directory filter cannot be efficiently implemented on object storage, and thus we should not publicly expose it.

Reply:
Yeah I did not expect the v2 API either until I reached a certain point in the changes yesterday and realized the current design requires it if we want a client to be able to use the directory filter. At that point adding the v2 the C API was a small change to test the work I had already done.. so I gave it a shot :shrug:

I am leaning towards just removing the directory filter and using the single callback. I'm going to give it a shot this morning and see how it goes, personally I'm in favor of not having a v2 API and simplifying some of the changes here. The only downside is without an API change I don't think we can toggle prefix collection behavior, unless we add a config parameter.